pages/proposals/index.vue

```diff
@@ -210,17 +221,47 @@ async function onCastBatchVotes() {
   const isDelegateeStored = isDelegatee.value;
 
   try {
+    let hash;
+    const reasons = selectedCastProposals.value.map((p) => p.reason || "");
     const proposalIds = selectedCastProposals.value.map((p) =>
       BigInt(p.proposalId),
     );
     const votes = selectedCastProposals.value.map((p) => p.vote);
 
-    const hash = await writeStandardGovernor(wagmiConfig, {
-      address: ttg.contracts.standardGovernor as Hash,
-      functionName: "castVotes",
-      args: [proposalIds, votes], // uint256 proposalId, uint8 support
-      account: userAccount.value,
-    });
+    const isOnlyOneVote = selectedCastProposals.value.length === 1;
+    const anyProposalHasReason = selectedCastProposals.value.some(
+      (p) => p.reason,
+    );
+
+    if (anyProposalHasReason && isOnlyOneVote) {
```

Comment:
this can be better written but the most readable, wanting comments

Reply:
maybe wrap the content of each if into a function just for better readability

Reply:
hash = await castVote(...all stuf)

casVote implementation the ifs

castVotes(){
if() castVoteWithReason
else if() castVote
...

}